File: AdventOfCodeCPlusPlus/AdventOfCodeCPlusPlus/InverseCaptcha.cpp
#include <string>;
#include "InverseCaptcha.h";

using namespace std;

InverseCaptcha::InverseCaptcha(void)
{
}

int InverseCaptcha::CalculateInverseCaptcha(string input)
{
	int sum = 0;
	for (int i = 0; i < input.length(); i++) 
	{
		int currentValue = input[i];
		int nextValue;
		if (i == input.length() - 1) 
		{
			nextValue = input[0];
		}
		else
		{
			nextValue = input[i + 1];
		}

		if (currentValue == nextValue)
		{
			sum += (int)currentValue - (int)'0';
		}
	}

	return sum;
}

int InverseCaptcha::CalculateInverseCaptchaPartTwo(string input)
{
	int sum = 0;
	for (int i = 0; i < input.length(); i++)
	{
		int currentValue = input[i];
		int desiredIndex = i + input.length() / 2;
		int nextValue;
		if (desiredIndex <= input.length() - 1)
		{
			nextValue = input[desiredIndex];
		}
		else
		{
			nextValue = input[desiredIndex - input.length()];
		}

		if (currentValue == nextValue)
		{
			sum += (int)currentValue - (int)'0';
		}
	}

	return sum;
}

File: AdventOfCodeJava/src/com/adventOfCode/InverseCaptcha.java
package com.adventOfCode;

public class InverseCaptcha {
	private InverseCaptcha() {
	}
	
    public static int calculateInverseCaptcha(String input){
        int sum = 0;

        for(int i = 0; i < input.length(); i++){
            int currentValueAsInt = Character.getNumericValue(input.charAt(i));
            int nextIndex = i == input.length() - 1 ? 0 : i + 1;
            int nextValueAsInt = Character.getNumericValue(input.charAt(nextIndex));

            if (currentValueAsInt == nextValueAsInt){
                sum += currentValueAsInt;
            }
        }

        return sum;
    }

    public static int calculateInverseCaptchaPartTwo(String input){
        int sum = 0;

        for(int i = 0; i < input.length(); i++){
            int currentValueAsInt = Character.getNumericValue(input.charAt(i));
            int desiredIndex = (i + input.length() / 2);
            int nextIndex = desiredIndex > input.length() - 1 ? desiredIndex - input.length() : desiredIndex;
            int nextValueAsInt = Character.getNumericValue(input.charAt(nextIndex));

            if (currentValueAsInt == nextValueAsInt){
                sum += currentValueAsInt;
            }
        }

        return sum;
    }
}
